prog.go

```diff
@@ -35,14 +35,54 @@ const (
 // verifier log.
 const DefaultVerifierLogSize = 64 * 1024
 
+// LogLevel controls the verbosity of the kernel's eBPF program verifier.
+// These constants can be used for the ProgramOptions.LogLevel field.
+type LogLevel uint32
+
+const (
+	// Print verifier state at branch points.
+	LogLevelBranch LogLevel = iota + 1
+
+	// Print verifier state for every instruction.
+	// Available since Linux v5.2.
+	LogLevelInstruction
+
+	// Print verifier errors and stats at the end of the verification process.
+	// Available since Linux v5.2.
+	LogLevelStats
+
+	// Disable the verifier log.
+	LogLevelDisabled = math.MaxUint32
```

Comment:
This is not kernel API right? What if we add `DisableDetailedErrors bool` to ProgramOptions instead?

Reply:
I get the rationale behind making the verifier log an implementation detail for future extensibility, but I'm not sure if that's really useful here. If we introduce a 'detailed errors' concept, that will need to be defined and explained to the caller. If the user is reaching for the option, it's mainly going to be for performance reasons, and they're going to be looking for a setting related to the verifier log.

Wouldn't `DisableVerifierLog bool` be more clear?

Reply:
Going with `LogDisabled` to stick to the `LogXYZ` pattern of field names.